src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -404,6 +460,30 @@ protected virtual void ValidateKeySize(SecurityKey key, string algorithm)
                 return;
             }
 
+            if (SecurityAlgorithms.Aes128Gcm.Equals(algorithm, StringComparison.Ordinal))
```

Comment:
Above in GetKeyBytes, we could improve this by calling JsonWebKeyConverter.TryConvertToSymmetricSecurityKey, this first checks if the ConvertedSecurityKey is set. This results in the key only being decoded once.

Reply:
Makes sense. Thanks for additional detail on this. I updated the PR with this change.